app/controllers/spree/credit_cards_controller.rb

```diff
@@ -63,16 +66,10 @@ def remove_shop_authorizations
 
     # It destroys the whole customer object
     def destroy_at_stripe
-      stripe_customer = Stripe::Customer.retrieve(@credit_card.gateway_customer_profile_id, {})
+      Stripe::CreditCardCloneDestroyer.new.destroy_clones(@credit_card)
 
-      stripe_customer&.delete
-    end
-
-    def stripe_account_id
-      StripeAccount.
-        find_by(enterprise_id: @credit_card.payment_method.preferred_enterprise_id).
-        andand.
-        stripe_user_id
+      stripe_customer = Stripe::Customer.retrieve(@credit_card.gateway_customer_profile_id, {})
+      stripe_customer&.delete unless stripe_customer.deleted?
```

Comment:
Since `stripe_customer ` can be `nil`, the `deleted?` call should be prefixed by the safe navigation operator as well.

Reply:
Fixed in https://github.com/openfoodfoundation/openfoodnetwork/pull/6522

Reply:
good catch @arku !